guides-markdown/cltv.md

```diff
@@ -0,0 +1,616 @@
+we can keep a reference to the redeem script, locking address (P2WSH address),
+locktime, and the redeem address. We'll do this by saving this information
+to a json in a separate text file.
+
+For this portion of the code, we're going to have two evaluation branches.
+We'll check if we have this file with information saved to it. If we don't,
+then we need to make the locking transaction. If we do, then we check if the current
+block height meets the locktime requirement (saved in our document) and
+create, sign, and broadcast our redeeming transaction
+
+
+```javascript
+async function lockAndRedeemCLTV(walletId) {
+  try {
+    const txInfoPath = './tx-info.json'; // this is where we'll persist our info
+    const wallet = walletClient.wallet(walletId); // instantiate a client for our wallet
+
+    let redeemScript, lockingAddr, locktime;
+
+    // check if file exists and if there is info saved to it
+    let txInfo = fs.existsSync(txInfoPath) ? fs.readFileSync(txInfoPath) : '';
+    if (!txInfo.length) {
+      // No saved transaction, so let's create it and then
+      // save the information for later
+
+      // Step 1: Setup wallet client and confirm balance
+      const { balance } = await wallet.getInfo();
+      assert(balance.confirmed > amountToFund.toValue(), 'Not enough funds!');
+
+      // Step 2: Setup keyring w/ pkh and create locking address
+      // that can be redeemed by our real wallet after a set locktime
+      const { publicKey, address } = await wallet.createAddress('default');
+
+      // create the keyring from the public key
+      // and get the public key hash for the locking script
+      const keyring = KeyRing.fromKey(Buffer.from(publicKey, 'hex'), true);
+      keyring.witness = true;
+      const pkh = keyring.getKeyHash();
+
+      // Get current height and set locktime to 10 blocks from now
+      const { chain: { height }} = await nodeClient.getInfo();
+      locktime = height + 10;
+
+      // create the script and address that can be redeemed by our wallet
+      redeemScript = createScript(locktime.toString(), pkh);
+      lockingAddr = getAddress(redeemScript, network);
+
+      // Step 3: use the wallet client to send funds to the locking address
+      const output = {
+        value: amountToFund.toValue(),
+        address: lockingAddr
+      };
+
+      const lockedTx = await wallet.send({ outputs: [output], rate: 7000 });
+      console.log('transaction sent to mempool');
+
+      // save the transaction information to to a file
+      txInfo = { lockedTx, lockingAddr, redeemScript, locktime, redeemAddress: address };
+      fs.writeFileSync(txInfoPath, JSON.stringify(txInfo, null, 2));
+
+      // mine one block to get tx on chain
+      // make sure you're doing this on regtest or simnet and not testnet or mainnet
+      // you can also use bPanel and the @bpanel/simple-mining plugin to do this instead
+      const minedBlock = await nodeClient.execute('generate', [1]);
```

Comment:
dunno if you need to mention that `rpc generate` will error if `--coinbase-address` isn't set at launch.